test/optimizer/stats_catalog_test.cpp

```diff
@@ -0,0 +1,220 @@
+#include <utility>
+
+#include "catalog/catalog.h"
+#include "catalog/postgres/pg_statistic.h"
+#include "gtest/gtest.h"
+#include "main/db_main.h"
+#include "optimizer/statistics/stats_storage.h"
+#include "parser/expression/constant_value_expression.h"
+#include "test_util/test_harness.h"
+#include "type/transient_value_factory.h"
+
+namespace terrier::optimizer {
+struct StatsCatalogTests : public TerrierTest {
+  void SetUp() override {
+    db_main_ = terrier::DBMain::Builder().SetUseGC(true).SetUseCatalog(true).Build();
+    txn_manager_ = db_main_->GetTransactionLayer()->GetTransactionManager();
+    catalog_ = db_main_->GetCatalogLayer()->GetCatalog();
+
+    auto *txn = txn_manager_->BeginTransaction();
+    db_ = catalog_->GetDatabaseOid(common::ManagedPointer(txn), catalog::DEFAULT_DATABASE);
+
+    // Ensure pg_statistic table exists with correct OID
+    auto accessor = catalog_->GetAccessor(common::ManagedPointer(txn), db_);
+    EXPECT_NE(accessor, nullptr);
+    VerifyStatsCatalog(*accessor);
+
+    // Initialize pointers to table and index
+    statistics_ = accessor->GetTable(catalog::postgres::STATISTIC_TABLE_OID);
+    statistics_oid_index_ = accessor->GetIndex(catalog::postgres::STATISTIC_OID_INDEX_OID);
+
+    txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+  }
+
+  void VerifyStatsCatalog(const catalog::CatalogAccessor &accessor) {
+    auto ns_oid = accessor.GetNamespaceOid("pg_catalog");
+    EXPECT_EQ(ns_oid, catalog::postgres::NAMESPACE_CATALOG_NAMESPACE_OID);
+    auto table_oid = accessor.GetTableOid(ns_oid, "pg_statistic");
+    EXPECT_EQ(table_oid, catalog::postgres::STATISTIC_TABLE_OID);
+  }
+
+  void GetStatTuplesForTable(common::ManagedPointer<transaction::TransactionContext> txn,
```

Comment:
It would be helpful to desribe what exactly does the function do and why you are passing invalid column id in the second column of the projected row. I understand it after reading it, but the API should be explained